common/src/ray_tracing.rs

```diff
@@ -0,0 +1,105 @@
+use crate::math;
+
+/// Performs ray tracing against the z=0 plane in node coordinates. This plane is one-sided, so the ray must be going in the -z
+/// direction to detect an intersection.
+///
+/// The `start_node_transform` parameter determines which coordinate system the `ray` parameter and any resulting intersection
+/// normals are given in. Specifically, the `start_node_transform` matrix converts this coordinate system to node coordinates.
```

Comment:
If we separated these concerns, I don't think it would save that much work. The reason why `ray_tracing` handles these transformations might not immediately be clear until the graph traversal commit, when there are multiple `node_transform`s.

Reply:
Checks out.